Editor/SnapZone/SnapZoneSettings.cs

```diff
@@ -161,16 +179,25 @@ private Material SetupValidationMaterial()
         {
             if (validationMaterial == null)
             {
-                validationMaterial = CreateMaterial();
-                validationMaterial.name = "SnapZoneValidationMaterial";
-
                 if (Directory.Exists(MaterialsPath) == false)
                 {
                     Directory.CreateDirectory(MaterialsPath);
                 }
+                
+                string validationMaterialName = "SnapZoneValidationMaterial";
+                string filePath = $"{MaterialsPath}/{validationMaterialName}.mat";
 
-                AssetDatabase.CreateAsset(validationMaterial, $"{MaterialsPath}/{validationMaterial.name}.mat");
-                AssetDatabase.Refresh();
+                if (File.Exists(filePath) == false) 
```

Comment:
Code duplication. Can it be written in a single method?

Reply:
Yes, did it. 👍